BiometricLoginKotlin/app/src/main/java/com/example/biometricloginsample/LoginActivity.kt

```diff
@@ -0,0 +1,190 @@
+    private val TAG = "LoginActivity"
+    private lateinit var biometricPrompt: BiometricPrompt
+    private val cryptographyManager = CryptographyManager()
+    private lateinit var encryptedServerTokenWrapper: CiphertextWrapper
+    private lateinit var binding: ActivityLoginBinding
+    private val loginWithPasswordViewModel by viewModels<LoginWithPasswordViewModel>()
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        binding = ActivityLoginBinding.inflate(layoutInflater)
+        setContentView(binding.root)
+    }
+
+    override fun onResume() {
+        super.onResume()
+        val ciphertextWrapper = cryptographyManager.getCiphertextWrapperFromSharedPrefs(
+            applicationContext,
+            SHARED_PREFS_FILENAME,
+            Context.MODE_PRIVATE,
+            CIPHERTEXT_WRAPPER
+        )
+
+        if (ciphertextWrapper != null) {
+            if (SampleAppUser.fakeToken == null) {
+                encryptedServerTokenWrapper = ciphertextWrapper
+                showBiometricPromptForDecryption()
+            } else {
+                // The user has already logged in, so proceed to the rest of the app
+                // this is a todo for you, the developer
+                updateApp(getString(R.string.already_signedin))
+            }
+        } else {
+            loginWithPasswordManager()
+        }
+
+        binding.useBiometrics.setOnClickListener {
+            if (ciphertextWrapper != null) {
+                encryptedServerTokenWrapper = ciphertextWrapper
+                showBiometricPromptForDecryption()
+            } else {
+                startActivity(Intent(this, EnableBiometricLoginActivity::class.java))
+            }
+        }
+    }
+
+    // BIOMETRICS SECTION
+
+    private fun showBiometricPromptForDecryption() {
+        val canAuthenticate = BiometricManager.from(applicationContext).canAuthenticate()
+        if (canAuthenticate == BiometricManager.BIOMETRIC_SUCCESS) {
+            val secretKeyName = getString(R.string.secret_key_name)
+            val cipher = cryptographyManager.getInitializedCipherForDecryption(
+                secretKeyName, encryptedServerTokenWrapper.initializationVector
+            )
+            biometricPrompt =
+                BiometricPromptUtils.createBiometricPrompt(this, ::decryptServerTokenFromStorage)
+            val promptInfo = BiometricPromptUtils.createPromptInfo(this)
+            biometricPrompt.authenticate(promptInfo, BiometricPrompt.CryptoObject(cipher))
+        }
+    }
+
+    private fun decryptServerTokenFromStorage(authResult: BiometricPrompt.AuthenticationResult) {
+        authResult?.cryptoObject?.cipher?.apply {
+            val plaintext =
+                cryptographyManager.decryptData(encryptedServerTokenWrapper.ciphertext, this)
+            SampleAppUser.fakeToken = plaintext
+            // now that you have the token, you can query server for everything else
```

Comment:
Nit: Sentences start with a capital letter.
```suggestion
            // Now that you have the token, you can query server for everything else
```